lms/views/admin/email.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from pyramid.httpexceptions import HTTPFound
+from pyramid.view import view_config, view_defaults
+
+from lms.security import Permissions
+from lms.services import DigestService, MailchimpService
+
+
+@view_defaults(route_name="admin.email", permission=Permissions.ADMIN)
+class AdminEmailViews:
+    def __init__(self, request):
+        self.request = request
+        self.digest_service = request.find_service(DigestService)
+        self.mailchimp_service = request.find_service(MailchimpService)
+
+    @view_config(
+        route_name="admin.email",
+        request_method="GET",
+        renderer="lms:templates/admin/email.html.jinja2",
+    )
+    def get(self):
+        return {}
+
+    @view_config(route_name="admin.email", request_method="POST")
+    def post(self):
+        to_email = self.request.POST["to_email"]
+        h_userids = self.request.POST["h_userids"].split()
+
+        # Caution: datetime.fromisoformat() doesn't support all ISO 8601
+        # strings! This only works for the subset of ISO 8601 produced by
+        # datetime.isoformat().
+        since = datetime.fromisoformat(self.request.POST["since"])
+        until = datetime.fromisoformat(self.request.POST["until"])
+
+        digests = self.digest_service.get_instructor_digest(h_userids, since, until)
+        self.mailchimp_service.send_instructor_digests(
+            digests, override_to_email=to_email
+        )
+
+        self.request.session.flash(
+            f"If the given users have any activity in the given timeframe then emails will be sent to {to_email}."
+        )
```

Comment:
The admin page doesn't actually know whether any emails will be sent: it depends on whether the given users have any activity in the given timeframe, and the admin page hasn't actually generated the digests it's only scheduled a celery task to generate them.

I think I do want the admin page to use the celery task though: it's more of an end-to-end test and it means that admin page form submissions won't timeout while the request tries to make possibly multiple calls to the Mailchimp API.